crates/vm/src/arch/aot/metered_execute.rs

```diff
@@ -341,16 +341,21 @@ where
         from_state: VmState<F, GuestMemory>,
         ctx: MeteredCtx,
     ) -> Result<(Vec<Segment>, VmState<F, GuestMemory>), ExecutionError> {
-        let vm_exec_state = VmExecState::new(from_state, ctx);
-        let vm_exec_state = self.execute_metered_until_suspend(vm_exec_state)?;
-        // handle execution error
-        match vm_exec_state.exit_code {
-            Ok(_) => Ok((
-                vm_exec_state.ctx.segmentation_ctx.segments,
-                vm_exec_state.vm_state,
-            )),
-            Err(e) => Err(e),
+        let mut exec_state = VmExecState::new(from_state, ctx);
+
+        loop {
```

Comment:
the loop does nothing because `execute_metered_until_suspend` never suspends in AOT, but it seemed better to just implement the general version defensively